mappings/net/minecraft/world/biome/Biome.mapping

```diff
@@ -36,6 +36,10 @@ CLASS net/minecraft/class_1959 net/minecraft/world/biome/Biome
 		ARG 1 pos
 	METHOD method_38104 (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/kinds/App;
 		ARG 0 instance
+	METHOD method_39927 canRain (Lnet/minecraft/class_2338;)Z
```

Comment:
This includes snow:

```suggestion
	METHOD method_39927 hasWeather (Lnet/minecraft/class_2338;)Z
```

Reply:
Does it? I'm not so sure about that. I think apple is correct

Reply:
The method is used just a few lines down in the `Biome#canSetSnow` method to prevent snow from being set in biomes without weather.

Reply:
It does not snow (nor freeze, see `canSetIce`) when it rains, and I do not think biomes with temperature above 0.15 makes them weather-free.

Reply:
Okay, `canRain` is technically misleading; it can return true in hot/weather-less biomes. However I believe it also returns true for biomes that can rain. While this does look ugly I do not think there are better unambiguous names.

```suggestion
	METHOD method_39927 doesNotSnow (Lnet/minecraft/class_2338;)Z
```